src/platform_impl/linux/wayland/window.rs

```diff
@@ -335,7 +335,7 @@ impl Window {
     pub fn fullscreen(&self) -> Option<Fullscreen> {
         if *(self.fullscreen.lock().unwrap()) {
             Some(Fullscreen::Borderless(RootMonitorHandle {
-                inner: PlatformMonitorHandle::Wayland(self.current_monitor()),
+                inner: PlatformMonitorHandle::Wayland(self.current_monitor().unwrap()),
```

Comment:
This is Wayland, right? What if no monitor is connected? The point of making current_monitor an option is because it might be None on Wayland, right?

Reply:
If you're in a fullscreen you must be presented on some output, however the unwrap part should be solved here https://github.com/rust-windowing/winit/pull/1670.

If nothing is connected, the app is fullscreened, and user calls to that function it'll likely crash, but it would crash earlier as well. There's no way to solve  that unless I rework the API in #1670.

Reply:
I mean there definitely is a way. It could just return None. That would actually be less code than unwrapping.

Reply:
yeah, but it'll be miss informing the users on a state, since you're in a fullscreen state actually, so I don't really like returning `None`.

Reply:
Internally we think that the user is in a fullscreen state, but Wayland is telling us that we're not. I think I'd rather trust Wayland here than potentially crashing? It just seems like it would be the desired behavior, even in case things do go wrong.

With stuff like this it's also always easy to have one end (like the fullscreen lock) messed up due to refactorings and then have another place suddenly crash. I don't really see the **advantage** of unwrapping.

Reply:
>Internally we think that the user is in a fullscreen state, but Wayland is telling us that we're not. I think I'd rather trust Wayland here than potentially crashing? It just seems like it would be the desired behavior, even in case things do go wrong.

No, it's not like that.

Wayland says that you **are** fullscreened, but it doesn't know on which monitor, since none are presented. And you can actually ask it to un fullscreen yourself or request fullscreen by passing `0` or calling `unset_fullscreen` respectively. So it's not that we think internally, we're saying that we're in fullscreen for sure, since Wayland told us so, but we can't say where, since every out disappeared.

The `self.fullscreen` mutex is set from the Wayland event in a callback handler, not from any of user events. If you do `set_fullscreen` on Wayland you're not touching this variable, it'll be touched when compositor tell you that you're in fullscreen that.

So performing an unwrap here is correct, since you know that you're in fullscreen, but you can't tell where. Returning `None` will lie to a user that you're not fullscreened, when in reality you're fullscreened.

>I think I'd rather trust Wayland here than potentially crashing?

Yes, that's what I'm doing. Wayland tells me that I'm fullscreened, but it doesn't know where, since the surface left all outputs. And since I can't expose that I'm just fullscreened and I must provide some output I crash, since there's no way to expose that unless #1670. 

>With stuff like this it's also always easy to have one end (like the fullscreen lock) messed up due to refactorings and then have another place suddenly crash. I don't really see the advantage of unwrapping.

You can't mess up this lock since you can only read it in winit, the set is done in handling of informing that your window actually went fullscreen. If you mess up the set of that mutex, you're already doing things incorrectly.

Reply:
> So performing an unwrap here is correct, since you know that you're in fullscreen, but you can't tell where. 

How is performing an unwrap correct? It's never correct to crash.

> Yes, that's what I'm doing. Wayland tells me that I'm fullscreened, but it doesn't know where, since the surface left all outputs. And since I can't expose that I'm just fullscreened and I must provide some output I crash, since there's no way to expose that unless #1670.

Crashing is **not** a reasonable response to being unable to determine which output is currently in fullscreen. If that could ever happen, crashing is completely unacceptable.

Reply:
And what should we do? lie to the users? postpone  PR? we're already crashing on master, so I don't see any issue with it for now, knowing that PR solving that, is already up(even in a WIP state though).

If you want to report incorrect value to a user, it's likely will be fine, but it's also wrong.

Reply:
"Lying" to the user is the much better alternative, yes. If we don't have a decent value to give to the user and the user cannot handle it either, it's much better to return an incorrect value that will allow the error to correct itself in the future rather than just crashing.

There's something that is more wrong than reporting an incorrect value and that's just straight-up crashing in a scenario where we have no good reason to crash.